pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -2779,6 +2781,10 @@ public class ServiceConfiguration implements PulsarConfiguration {
     )
     private Set<String> additionalServlets = new TreeSet<>();
 
+    public boolean isSubscriptionKeySharedEnable() {
+        return subscriptionKeySharedEnable && subscriptionTypesEnabled.contains("Key_Shared");
+    }
```

Comment:
Here should be a bug fix. If the subscriptionKeySharedEnable = true but the subscriptionTypesEnabled doesn't have the `Key_Shared` subscription, we should treat it as Key_Shared subscription disabled.

https://github.com/apache/pulsar/blob/b68415a89dcd067511d01956447d1a507865534c/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java#L704-L708

Could you please add a UT?

Reply:
I'm glad to do so. Shall I separate the bug fix into a new PR or join in this PR?

Reply:
@codelipenghui I'll push a commit to add a UT along with this patch later. Please confirm whether it's proper when you have spare time :)

Reply:
Updated. Please give it another pass of review.

cc @codelipenghui